schema.ts

```diff
@@ -905,6 +920,12 @@ export type ObjectSchemaProperties<K extends string = never> = {
 /** @hidden */
 export type GenericObjectSchema = ObjectSchema<string, string>;
 
+/**
+ * When an object schema is being used as a property of another object, then it additionally has
+ * ObjectSchemaProperty properties.
+ */
+export type PropertyObjectSchema = GenericObjectSchema & ObjectSchemaProperty;
```

Comment:
I would probably either skip this, or create `PropertyBooleanSchema`, `PropertyNumberSchema`, etc as well.